src/elements/renderRequestButton.ts

```diff
@@ -0,0 +1,214 @@
+  )
+  text.setAttribute('fill', '#fff')
+
+  svg.appendChild(text)
+
+  return svg
+}
+/* tslint:enable:max-line-length */
+
+const renderTextAndLogo = (id: string) => {
+  const textAndLogo = document.createElement('span')
+  textAndLogo.id = `${id}-text-and-logo`
+  textAndLogo.appendChild(renderBloomLogo(id))
+  textAndLogo.appendChild(renderText(id))
+
+  return textAndLogo
+}
+
+const getLink = (data: RequestData) => `https://bloom.co/download?request=${window.btoa(JSON.stringify(data))}`
+
+const renderRequestButton = (config: {container: HTMLElement; requestData: RequestData}): RequestButtonResult => {
+  const id = generateId()
+
+  const anchor = document.createElement('a')
+  anchor.id = id
+
+  anchor.href = getLink(config.requestData)
+  anchor.target = '_blank'
+  anchor.rel = 'norefferer noopener'
+
+  anchor.appendChild(renderStyle(id))
+  anchor.appendChild(renderLockIcon(id))
+  anchor.appendChild(renderTextAndLogo(id))
+
+  config.container.appendChild(anchor)
+
+  return {
```

Comment:
So, by generating an id and applying it to the `update` and `remove` returned to the client, we facilitate the markup management for the client? I like it! 💯

Reply:
Yup! consumer doesn't need to keep track of the id to update/remove the element. Just have to hold on to the return value